yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,334 @@
+                l = (
+                    int(file[-5:-3]) - 1
+                )  # Pulls the subgrid number from the data file name
+                if nmz > 2:
+                    k = f["fluid"]["entropy"].shape[0]
+                    r = f["mesh"]["x_ef"][:-2]
+                    theta = f["mesh"]["y_ef"][:]
+                    phi = f["mesh"]["z_ef"][
+                        k * l : k * (l + 1) + 1
+                    ]  # Pulls only the individual subgrid's band of phi values
+
+                else:
+                    r = f["mesh"]["x_ef"][:]
+                    theta = f["mesh"]["y_ef"][:]
+                    phi = f["mesh"]["z_ef"][:]
+
+                # Creates variables to hold the size of dimensions
+                nxd = r.size
+                nyd = theta.size
+                nzd = phi.size
+                nyzd = nyd * nzd
+                nyzd_ = (nyd - 1) * (nzd - 1)
+
+                # Generates and fills coordinate array
+                coords = np.zeros((nxd, nyd, nzd, 3), dtype="float64", order="C")
+                coords[:, :, :, 0] = r[:, None, None]
+                coords[:, :, :, 1] = theta[None, :, None]
+                coords[:, :, :, 2] = phi[None, None, :]
+                coords.shape = (nxd * nyd * nzd, 3)
+
+                if yy:
+                    # Converts coordinates to cartesian in the case of Yin-Yang data
+                    x = coords[:, 0] * np.sin(coords[:, 1]) * np.cos(coords[:, 2])
+                    y = coords[:, 0] * np.sin(coords[:, 1]) * np.sin(coords[:, 2])
+                    z = coords[:, 0] * np.cos(coords[:, 1])
+
+                    if (
+                        "grid_2" in file
+                    ):  # Applies transform to meshes in the rotated grid
+                        x_hold = -x
+                        y_hold = z
+                        z_hold = y
+                    else:
+                        x_hold = x
+                        y_hold = y
+                        z_hold = z
+
+                    coords[:, 0] = x_hold
+                    coords[:, 1] = y_hold
+                    coords[:, 2] = z_hold
+
+                    """
+                    Spherical transform -- Currently Unused
+                    old_theta = coords[:,1].copy()
+                    old_phi = coords[:,2].copy()
+                    coords[:,1] = np.arccos(np.sin(old_theta)*np.sin(old_phi))
+                    nu = np.cos(old_theta)
+                    de = (-1.0*np.sin(old_theta))*np.cos(old_phi)
+                    coords[:,2] = np.arctan2(nu,de)
+                    """
+
+                """
+                Connectivity is an array of rows, each of which corresponds to a grid cell. The 8 elements of each row are
+                integers representing the cell verticies. These integers refrence the numerical index of the element of the
+                "coords" array whcih corresponds to the spatial coordinate.
+                """
```

Comment:
Please use actual comments. We do not use multiple lines string literals as comments in yt

Reply:
Spherical transforms removed and Connectivity text changed to comments in 10df6b4